### PR TITLE
fix mutation over websocket/subscription

### PR DESCRIPTION
**Pull request by tomi-bigpi**
Fixes issue where mutations are not allowed over web sockets (subscriptions).

See: https://github.com/mercurius-js/mercurius/discussions/977

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -471,7 +471,7 @@ const plugin = fp(async function (app, opts) {
       document = cached.document
     }
 
-    if (reply && reply.request.raw.method === 'GET') {
+    if (reply && reply.request.raw.method === 'GET' && reply.request.headers.upgrade?.toLowerCase() !== 'websocket') {
```
Comment (mcollina) [inline index.js:474]:
```suggestion
    if (reply && reply.request.raw.method === 'GET' && !reply.request.ws) {
```

Comment (mcollina) [reply index.js:474]: This should work, check it out!

Comment (tomi-bigpi) [reply index.js]: Thanks, it worked with manual testing, so I updated the corresponding test to set that flag on the request like `fastify-websocket` does.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**